### ISSUE TITLE
Support 'lastupdated' metadata for blog posts

### ISSUE THREAD

**Issue by StarfallProjects** [squidfunk/mkdocs-material#5028]
### Context

This is for my personal projects, not n8n work (so doesn't qualify for any prioritization)

### Description

The Insiders theme supports the following:

1. Add `date` to page frontmatter
2. Set [preferred format](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#posts) in mkdocs.yml
3. Date is displayed on each blog post

It would be great to also support a `lastupdated` date option. 

### Related links

https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.post_date_format

### Use Cases

Allows users to see if a blog post has been updated since publishing. This is quite important, especially for technical info which can become obsolete fast. Having a way to signal a post has been updated allows writers to keep content looking fresh / to reassure readers that content is still relevant.

I explored just overriding the blog-post.html template. It's easy to display any frontmatter I choose, but getting the date nicely formatted stumped me. It looks like the theme does quite a lot of heavy lifting here. I played around with strftime and strptime and _both_ failed with apparently contradictory errors. Hence requesting built in support.

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material), related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Thanks for suggesting. Date formatting is done in the plugin's context (not the template) using `babel.dates.format_date`, because different languages come with different date formats, so `strftime` is not enough 😊 We'll consider this extension as the next iteration on the blog plugin and template.

**Comment by bilal-fazlani** [comment]
It would be useful to have this as a metadata 

I am thinking of adding custom scripts for SEO for blog articles, where last updated can be used by google. 
its not mandatory I think but might be useful to tell google that this article is updated with up to date information now


https://developers.google.com/search/docs/appearance/structured-data/article#examples

**Comment by jonaharagon** [comment]
I didn't have an issue using `strftime` in the post template to achieve this: https://github.com/privacyguides/blog.privacyguides.org/blob/main/theme/blog-post.html#L74-L84 ([demo](https://blog.privacyguides.org/2022/07/07/signal-configuration-and-hardening/))

...unless I misunderstand the issue.

**Comment by squidfunk** [comment]
`strftime` doesn't give you formatting in locales other than English, except for when you change `LC_ALL`. This is why we use `babel` to format dates and times, as it support all locales without modifying the environment.

**Comment by squidfunk** [comment]
Added in 04bde62c9. Note that this is currently only on the [`merge/piri-piri`](https://github.com/squidfunk/mkdocs-material/pull/5683) branch that is the base branch for 9.2.0b2. As soon as we finalized the beta and release the stable version, we'll also merge this back to Insiders. The reason why we're waiting is that we need to adjust the blog plugin to MKDocs 1.5 which changed some internal APIs that make some things simpler and more stable, especially interop with other plugins like awesome-pages.

Here's how it looks:

<img width="215" alt="Bildschirmfoto 2023-08-01 um 19 05 49" src="https://github.com/squidfunk/mkdocs-material/assets/932156/0d912210-f976-49e6-904d-5db5f86fd716">

And here's how you use it:

``` markdown
---
date: 2023-07-30
date_updated: 2023-08-01
---

# Post title
...
```

We'll also likely refactor the styling a little before going final.

**Comment by squidfunk** [comment]
__Change of plans__: added upstream in [`13cf94776`](https://github.com/squidfunk/mkdocs-material-insiders/commit/13cf947761630dc99a83dacc28e726ccb1da887b). We decided to upstream this before the 9.2 release, because we want to put it in the hands of Insiders users right now, because there's no point in letting our sponsors wait 🚀

**Comment by squidfunk** [comment]
Released as part of [9.1.21+insiders-4.39.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.39.0).

**Comment by squidfunk** [comment]
[We changed the syntax a little](http://localhost:8000/mkdocs-material/setup/setting-up-a-blog/#writing-your-first-post), so authors can add custom dates that they can add to the blog templates:

<img width="708" alt="Bildschirm­foto 2023-08-21 um 18 59 39" src="https://github.com/squidfunk/mkdocs-material/assets/932156/14519042-169b-4356-ae42-eb569c4ed3e2">
